spec/factories/api_tokens.rb

```diff
@@ -0,0 +1,11 @@
+FactoryBot.define do
+  factory :api_token do
+    lead_provider { create(:lead_provider) }
+    hashed_token { Devise.token_generator.digest(APIToken, :hashed_token, "token") }
+
+    trait :teacher_record_service do
+      scope { APIToken.scopes[:teacher_record_service] }
+      hashed_token { Devise.token_generator.digest(APIToken, :hashed_token, "trs_token") }
+    end
```

Comment:
The developer needs to be able to either predict the raw token or control it, moving it to a transient block allows for generation of multiple tokens of the same scope in the future?

```ruby
    transient do
      raw_token { "token" }
    end

    lead_provider { create(:lead_provider) }
    hashed_token { Devise.token_generator.digest(APIToken, :hashed_token, raw_token) }

    trait :teacher_record_service do
      raw_token { "trs_token" }

      lead_provider { nil }
      scope { APIToken.scopes[:teacher_record_service] }
    end
```